xwiki-commons-core/xwiki-commons-displayer/src/main/java/org/xwiki/displayer/internal/DefaultHTMLDisplayerManager.java

```diff
@@ -105,19 +105,31 @@
     @Override
     public <T> String display(Type targetType, T value) throws HTMLDisplayerException
     {
-        return getHTMLDisplayer(targetType).display(value);
+        if (targetType == null && value != null) {
```

Comment:
Just a detail: I would have probably created a private (maybe it make sense even in public not sure) getHTMLDisplayer(Type targetType, Object value) to put this logic in a common method.

Reply:
Right, I'll refactor